scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,364 @@
+import qcore.simulation_structure as sim_struct
+import qcore.constants as const
+
+
+# The process types that are used for progress tracking
+PROCESS_TYPES = [
+    const.ProcessType.EMOD3D.str_value,
+    const.ProcessType.HF.str_value,
+    const.ProcessType.BB.str_value,
+]
+
+EST_CORE_HOURS_COL = "est_core_hours"
+ACT_CORE_HOURS_COL = "act_core_hours"
+
+R_COUNT_COL = "r_count"
+
+SLACK_ALERT_CONFIG = "slack_alert.yaml"
+
+
+def get_chours(json_file, proc_types: List[str], debug=False):
+    """Gets the core hours for the specified process types (i.e. EMOD3D, HF, BB, ...)
+    for the specified simulation metadata json log file.
+    """
+    with open(json_file, "r") as f:
+        data = json.load(f)
+    sim_name = data.get(const.MetadataField.sim_name.value)
+    core_hours = []
+    completed_proc = {}
+    for proc_type in proc_types:
+        if proc_type in data.keys():
+            core_hours.append(
+                data[proc_type].get(const.MetadataField.run_time.value, np.nan)
+                * data[proc_type].get(const.MetadataField.n_cores.value, np.nan)
+                / 3600.0
+            )
+
+        else:
+            core_hours.append(np.nan)
+    return sim_name, core_hours
+
+
+def get_chours_used(fault_dirs: Iterable[str]):
+    """Returns a dataframe containing the core hours used for each of the
+    specified faults"""
+    faults, core_hours, completed_r_counts, missing_data = [], [], [], []
+    for fault in fault_dirs:
+        fault = Path(fault)
+        fault_name = fault.stem
+        cur_fault_chours = np.asarray(
+            [
+                get_chours(json_file, PROCESS_TYPES)[1]
+                for json_file in fault.glob(f"**/{const.METADATA_LOG_FILENAME}")
+            ]
+        ).reshape(-1, len(PROCESS_TYPES))
+
+        faults.append(fault_name)
+        if cur_fault_chours.shape[0] > 0:
+            core_hours.append(np.nansum(cur_fault_chours, axis=0))
+        else:
+            core_hours.append([np.nan, np.nan, np.nan])
+
+    df = pd.DataFrame(index=faults, data=core_hours, columns=PROCESS_TYPES)
+
+    return df
+
+
+def get_completed_r_count(db_file, proc_type: str, fault_name: str = None):
+    """Returns the number of realisations of a fault "fault_name" that completed the proc_type. If fault_name is None,
+    consider all faults """
+    proc_type_int = const.ProcessType[proc_type].value
+    conn = sqlite3.connect(str(db_file))
+    c = conn.cursor()
+    if fault_name is None:
+        c.execute(
+            f"SELECT COUNT(*) FROM state WHERE proc_type=? and status={const.Status['completed'].value}",
+            (proc_type_int,),
+        )
+    else:
+        c.execute(
+            f"SELECT COUNT(*) FROM state WHERE proc_type=? and status={const.Status['completed'].value} and run_name LIKE ?",
+            (proc_type_int, fault_name + "_REL%"),
+        )
+
+    completed_r_count = c.fetchone()[0]
+    conn.close()
+    return completed_r_count
```

Comment:
Is it possible to use the mgmt db class for this?

Reply:
Done this - some changes had to be made to MgmtDB.py